velox/functions/lib/Re2Functions.cpp

```diff
@@ -767,6 +926,36 @@ std::shared_ptr<exec::VectorFunction> makeLike(
       name,
       inputArgs[1].type->toString());
   auto pattern = constantPattern->as<ConstantVector<StringView>>()->valueAt(0);
+  if (!escapeChar) {
+    PatternKind patternKind;
+    vector_size_t reducedLength;
+    std::tie(patternKind, reducedLength) = determinePatternKind(pattern);
+
+    switch (patternKind) {
+      case PatternKind::kExactlyN:
+        return std::make_shared<
+            OptimizedLikeConstantPattern<PatternKind::kExactlyN>>(
+            pattern, reducedLength);
+      case PatternKind::kAtLeastN:
+        return std::make_shared<
+            OptimizedLikeConstantPattern<PatternKind::kAtLeastN>>(
+            pattern, reducedLength);
+      case PatternKind::kFixed:
+        return std::make_shared<
+            OptimizedLikeConstantPattern<PatternKind::kFixed>>(
+            pattern, reducedLength);
+      case PatternKind::kPrefix:
+        return std::make_shared<
+            OptimizedLikeConstantPattern<PatternKind::kPrefix>>(
+            pattern, reducedLength);
+      case PatternKind::kSuffix:
+        return std::make_shared<
+            OptimizedLikeConstantPattern<PatternKind::kSuffix>>(
```

Comment:
Nit : Not to increase the work but this name is unnecessarily long as the pattern is always constant. Maybe LikeWithRe2 and OptimizedLikeWithMemcmp are better options ?